pvlib/irradiance.py

```diff
@@ -213,15 +230,21 @@ def aoi(surface_tilt, surface_azimuth, solar_zenith, solar_azimuth):
     aoi : numeric
         Angle of incidence in degrees.
     """
+    # see discussion in https://github.com/pvlib/pvlib-python/issues/1185
+    solar_vec = _spherical_to_cartesian(solar_zenith, solar_azimuth)
+    surface_vec = _spherical_to_cartesian(surface_tilt, surface_azimuth)
 
-    projection = aoi_projection(surface_tilt, surface_azimuth,
-                                solar_zenith, solar_azimuth)
-    aoi_value = np.rad2deg(np.arccos(projection))
+    c_vec = solar_vec - surface_vec
+    c_sqrd = np.sum((c_vec*c_vec).T, axis=0)
+    c = np.sqrt(c_sqrd)
 
-    try:
-        aoi_value.name = 'aoi'
-    except AttributeError:
-        pass
+    aoi = 2 * np.arctan(c / np.sqrt((1+(1+c))*((1-c)+1)))
+    aoi_value = np.rad2deg(aoi)
+
+    for arg in [surface_tilt, surface_azimuth, solar_zenith, solar_azimuth]:
```

Comment:
I think it's OK to assume that these Series have a common index, and stop the loop when one index if found.

Reply:
does the new implementation preserve Series in --> Series out without this code? 

I'm fine with dropping the name assignment and letting users deal with that.

Reply:
> does the new implementation preserve Series in --> Series out without this code?

Unfortunately it does not; the `_spherical_to_cartesian` function turns Series into 2-D numpy arrays.  I don't immediately see a way around it that's cleaner than the "fix types at the end" idiom.